pydm/display_module.py

```diff
@@ -1,40 +1,247 @@
+import functools
+import imp
+import inspect
+import logging
+import os
 import sys
+import uuid
+import warnings
 from os import path
+
 from qtpy import uic
-from qtpy.QtWidgets import QWidget
-from .utilities import macro
+from qtpy.QtWidgets import QWidget, QApplication
+
+from .utilities import macro, is_pydm_app
+
+
+class ScreenTarget:
```

Comment:
Consider using `class ScreenTarget(str, enum.Enum)` here so that `'new_process'` and `'dialog'` strings can be valid options.

Reply:
Still keeping Py27 for this PR... that is why I'm not using `enum`.